File: 7. Code Challenges/12. Staircase/staircase.py
def stairCase(n):
    cases = [list(str(i)) for i in range(1,int("1"*n)+1) \
                  if len(list(str(i))) >= round(n/2) \
                 and sum(map(int,list(str(i)))) == n \
                 and "0" not in list(str(i)) \
                 and "3" not in list(str(i)) \
                 and "4" not in list(str(i)) \
                 and "5" not in list(str(i)) \
                 and "6" not in list(str(i)) \
                 and "7" not in list(str(i)) \
                 and "8" not in list(str(i)) \
                 and "9" not in list(str(i))]
    
    print(f"Number of Unique Ways:  {len(cases)}\n")
    steps = 0
    for i in cases:
        steps += 1
        print(f"{steps}:\t{', '.join(i)}")

stairCase(5)

File: 7. Code Challenges/12. Staircase/staircase.js
function stairCase(n){
    let allPerms = "";
    for (let i = 0; i < n; i++) {allPerms += "1";}
    let start = Math.round(n/2);
    let cases = [];

    function sumofArray(array){
        let sum = 0;
        for (let i = 0; i < array.length; i++) {
            sum += +array[i];
        }
        return sum; 
    }
    for (let i = 1; i <= +allPerms; i++){
        let perms = String(i).split("");
            perms.length >= start && sumofArray(perms) == n
            && !perms.includes("0") && !perms.includes("3")  
            && !perms.includes("4") && !perms.includes("6")  
            && !perms.includes("7") && !perms.includes("8")  
            && !perms.includes("9")   
            ? cases.push(perms) : null;
        }
    console.log(`Number of Stairs: ${n}`);
    console.log(`Number of Unique Ways:  ${cases.length}`);
    let steps = 0;
    for (const i of cases) {
        steps++;
        console.log(`${steps}:\t ${i.join(", ")}`);
    }
}

stairCase(5);



